main/src/com/google/refine/importers/FixedWidthImporter.java

```diff
@@ -217,7 +214,7 @@ static public int[] guessColumnWidths(File file, String encoding) {
                 }
 
                 if (counts != null && lineCount > 2) {
-                    List<Integer> widths = new ArrayList<Integer>();
+                    List<Integer> widths = new ArrayList<>();
```

Comment:
I'd be super grateful if we can avoid mixing large-scale style changes and functional changes.
I have no problem with normalizing `new ArrayList<Integer>()` to `new ArrayList<>()` but I would do this in a separate PR (potentially alongside other cosmetic changes). Ideally we should also change the linter configuration for it to enforce this convention across the codebase (if possible).

Reply:
As a reminder, again, "Draft" means "not ready for review."

If you're looking for something to review, I marked #5397 as ready for review 3 days ago.

Reply:
I feel like it's useful to flag this sort of problem early on, before you bundle up more unrelated changes in the same PR, because they can be tricky to untangle later on. That is something that was a problem in past PRs, so it feels important to me to point this out.

If getting feedback at this stage is problematic for you, I wonder if you could just not open a PR yet - that would help us keep the backlog shorter. If the intention with opening a PR is to show that you are working on this to avoid someone else doing the same, you can just assign you to the issue, no? You could open the PR to run the CI, but in this case you are working on the backend so I trust you should be able to run the test suite locally without too much problem.

Another problem with draft PRs is that I do not actually get notifications when people mark them as ready for review - but I do get notifications when people open them. I wonder if this is something I can fix in my settings.

Also, I would be grateful if you could adopt a friendlier tone. Your comment above reads quite dry.

Reply:
> I feel like it's useful to flag this sort of problem early on, before you bundle up more unrelated changes in the same PR, because they can be tricky to untangle later on. 

That's why I (usually) segregate them in separate commits. There are also unrelated bug fixes for problems found by my IDE, also separated in their own commits.

> That is something that was a problem in past PRs, 

If you're talking about my PRs, I don't remember it being mentioned. Sorry if I missed it.

> You could open the PR to run the CI, but in this case you are working on the backend so I trust you should be able to run the test suite locally without too much problem.

As you can see from 093c4c212638ede3869cbf13470c5fc38938f18f, CI produces [different results](https://github.com/OpenRefine/OpenRefine/actions/runs/3493620663) from the test suite. If there's something I'm missing in my local setup, I'd welcome a link to the documentation.

If it's project policy that Draft PRs aren't allowed, I will comply, of course, but it'll probably mean fewer contributions.